### PR TITLE
add the numerical core page and outline of advanced numerical page

### PR DESCRIPTION
**Pull request by lcwheeler**

### CODE REVIEW

**Comment by Zsailer** [comment]
Thanks Luke!

Looks good. I'll do a read through this afternoon.

**Comment by lcwheeler** [comment]
Cool, sounds good! Thanks, Zach!

**Comment by lcwheeler** [comment]
Just added an updated version of the "Advanced numerical libraries" page. It's got descriptions of each included library. Let me know if you think there's something missing.

File: docs/_pages/advanced_numerical.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+To install this library in your Python environment simply run: 
+::
+
+    pip install keras
+
+LmFit
+-----
+
+LmFit is a powerful library for curve-fitting in Python. It implements several methods for non-linear least-squares minimization and introduces some very useful classes. The LmFit Parameters class offers a clean way of handling model paramters while fitting.
```
Comment (Zsailer) [inline docs/_pages/advanced_numerical.rst]: - Change description to something like: "LmFit is a great tool for non-linear curve fitting. It uses SciPy under-the-hood, but offers a better interface. Specifically, it offers more control when estimating model parameters. If you're going to be fitting complex data in Python, we suggest LmFit."

File: docs/_pages/advanced_numerical.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
+Advanced Numerical Libraries
+============================
+
+This page introduces you to a set of powerful Python libraries for advanced numerical computing. Python has libraries for machine learning, model fitting, statistics, network calculations, and much more! 
+
+scikit-learn
+------------
+
+scikit-learn is a powerful Python library for machine learning. It contains a wide array of useful tools for classification, regression, clustering, dimensionality reduction, and much more. Implementations are available for most standard machine learning algorithms, such as Support Vectro Machines and Decision Trees. If you're planning to implement machine learning in your analyses, then you will want to install scikit-learn.
```
[the remark this answers is not in the document]
Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1: Fixed!

File: docs/_pages/advanced_numerical.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+To install this library in your Python environment simply run: 
+::
+
+    pip install scikit-learn
+
+scikit-image
+------------
+
+scikit-image is a very useful Python library for handling image data. It's a powerful tool for image processing that allows users to identify objects, filter images, mainipulate color channels, and a wide variety of other tasks.
```
[the remark this answers is not in the document]
Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1: Fixed! I will push these changes to my fork soon and add to the pull request.

File: docs/_pages/advanced_numerical.rst (hunk 1 above)
Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1: Fixed!

File: docs/_pages/advanced_numerical.rst [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+To install this library in your Python environment simply run: 
+::
+
+    pip install lmfit
+
+emcee
+-----
+
+emcee is a Python library built for efficient Bayesian computations. It implements the Affine Invariant Markov chain Monte Carlo (MCMC) Ensemble sampler to reconstruct posterior probability distributions for model parameters. If you're looking to implement a Bayesian fit in your analysis this libary will be a useful tool.
```
Comment (Zsailer) [inline docs/_pages/advanced_numerical.rst]: - Second sentence is maybe too jargony?
- How about: "emcee is a Python library for efficiently estimating probability distributions. It uses efficient MCMC sampling strategy that is often used to approximate posterior distributions in Bayes Theorem. 

It uses a effici

Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1:

File: docs/_pages/advanced_numerical.rst [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+To install this library in your Python environment simply run: 
+::
+
+    pip install pymc3
+
+Networkx
+--------
+
+Networkx is an excellent library designed to deal with network analyses. It introduces a Graph datastructure that makes encoding networks easy and intuitive. Networkx includes a variety of tools for building and analyzing networks. It has the ability to export Graph objects in many different formats and can interface directly with NumPy and Pandas. If you're looking to deal with networks in your analysis Networkx will be extremely useful.
```
Comment (Zsailer) [inline docs/_pages/advanced_numerical.rst]: Change to? "Networkx offers a simple interface for managing network data. It introduces a Graph datatype that is easy and intuitive to use. It also provides various algorithms for analyzing and plotting networks."

File: docs/_pages/advanced_numerical.rst [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+Numba
+-----
+
+Numba is a library designed to help you speed up your Python calculations. It achieves this goal using a just-in-time compiler, which gives Python code speed that is comparable in performance to C. Numba is easy to use. Python functions can be wrapped with a simple decorator that results in increased speed. 
+
+To install this library in your Python environment simply run: 
+::
+
+    pip install scikit-learn numba
```
Comment (Zsailer) [inline docs/_pages/advanced_numerical.rst]: just numba?

Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1:

File: docs/_pages/advanced_numerical.rst (hunk 5 above)
Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1:

File: docs/_pages/advanced_numerical.rst [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+Cython
+------
+
+Cython is another library aimed at speeding up Python code. It effectively allows users to write C code using Python syntax. Cython code is compiled and runs at C speed, increasing the performance of your code.
```
Comment (Zsailer) [inline docs/_pages/advanced_numerical.rst]: Cython is another library aimed at speeding up Python code. Users can write Python code and quickly translate it to a C extension.

Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1:

File: docs/_pages/advanced_numerical.rst [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+To install this library in your Python environment simply run: 
+::
+
+    pip install cython   
+
+SymPy
+-----
+
+SymPy is a useful library for writing mathematical code. It allows users to write mathematical expressions in a simple symbolic syntax. If you're writing any complicated mathematical functions in your code SymPy will likely be useful to you.
```
Comment (Zsailer) [inline docs/_pages/advanced_numerical.rst]: SymPy is a library for doing symbolic math. You can compute integrals, derivatives, algebraic manipulations, etc. Think Mathematica in Python.

Comment (lcwheeler) [reply docs/_pages/advanced_numerical.rst]: :+1:

**Comment by Zsailer** [comment]
Everything looks good Luke! Great work!

**Comment by lcwheeler** [comment]
Thanks, Zach! I appreciate you reviewing all of that!

**Comment by Zsailer** [comment]
I'm thinking we should reorder the "advanced libraries". I think Theano, Keras, and Tensorflow are *really* advanced and should be at the bottom. 

I propose:

1. scikit-learn
2. scikit-image
3. lmfit
4. networkx
5. pymc3
6. statsmodels
7. astropy
6. cython
7. numba
8. tensorflow
9. theano
10. keras

**Comment by lcwheeler** [comment]
Okay I can work on that later.

**Comment by lcwheeler** [comment]
The advanced numerical libraries page has been reordered.

**Comment by Zsailer** [comment]
Awesome! Merging.

**Pull request merged**